### PR TITLE
Fold System.Globalization.Extensions contract to System.Runtime.Extensions

### PR DESCRIPTION
**Pull request by tarekgh**
This change also contains some clean up for the normalization source files and limit System.Private.Uri to use the normalization files in common folder only when building as uap101

### CODE REVIEW

**Comment by tarekgh** [comment]
#13804

**Comment by tarekgh** [comment]
CC: @weshaggard @jkotas @danmosemsft

File: src/System.Globalization.Extensions/src/System.Globalization.Extensions.csproj [hunk 1]
```diff
@@ -24,43 +24,13 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Release|AnyCPU'" />
```
Comment (jkotas) [inline src/System.Globalization.Extensions/src/System.Globalization.Extensions.csproj]: Could you please switch the build to reuse the same binaries for both Windows and Unix. We do not need different builds for this anymore.

File: src/System.Globalization.Extensions/tests/Normalization/StringNormalizationTests.cs [hunk 2]
```diff
@@ -31,7 +31,7 @@ public void IsNormalized_Invalid()
             Assert.Throws<ArgumentException>("strInput", () => "\uFFFE".IsNormalized()); // Invalid codepoint
             Assert.Throws<ArgumentException>("strInput", () => "\uD800\uD800".IsNormalized()); // Invalid surrogate pair
 
-            Assert.Throws<ArgumentNullException>("strInput", () => StringNormalizationExtensions.IsNormalized(null));
+            Assert.Throws<NullReferenceException>(() => StringNormalizationExtensions.IsNormalized(null));
```
Comment (jkotas) [inline src/System.Globalization.Extensions/tests/Normalization/StringNormalizationTests.cs]: Why not throw the same exception as before?

Comment (tarekgh) [inline src/System.Globalization.Extensions/tests/Normalization/StringNormalizationTests.cs]: I changed the implementation to just call String.IsNormalized() for simplicity and get same behavior as the string too. it you think this is a problem I can re-add the checks back

Comment (jkotas) [inline src/System.Globalization.Extensions/tests/Normalization/StringNormalizationTests.cs]: I believe that the general convention for extension methods has been to explicitly check for null.

Comment (tarekgh) [inline src/System.Globalization.Extensions/tests/Normalization/StringNormalizationTests.cs]: ok I will re-add the checks back

File: src/System.Globalization.Extensions/src/ApiCompatBaseline.net463.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+TypesMustExist : Type 'System.StringNormalizationExtensions' does not exist in the implementation but it does exist in the contract.
```
Comment (weshaggard) [inline src/System.Globalization.Extensions/src/ApiCompatBaseline.net463.txt]: Can you please add a comment similar to https://github.com/dotnet/corefx/blob/master/src/System.Security.SecureString/src/ApiCompatBaseline.txt#L2.

File: src/System.Runtime.Extensions/ref/System.Runtime.Extensions.cs [hunk 4]
```diff
@@ -948,6 +948,23 @@ public partial class ContextStaticAttribute : System.Attribute
     {
         public ContextStaticAttribute() { }
     }
+
```
Comment (weshaggard) [inline src/System.Runtime.Extensions/ref/System.Runtime.Extensions.cs]: nit: empty line.

**Comment by tarekgh** [comment]
I have addressed all comments (so far...)

File: src/System.Globalization.Extensions/src/System.Globalization.Extensions.csproj [hunk 5]
```diff
@@ -18,49 +18,17 @@
     <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
   </PropertyGroup>
   <!-- Help VS understand available configurations -->
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Unix_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Unix_Release|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'" />
```
Comment (jkotas) [inline src/System.Globalization.Extensions/src/System.Globalization.Extensions.csproj]: The change to unify Windows and Unix needs to be done in https://github.com/dotnet/corefx/blob/master/src/System.Globalization.Extensions/src/System.Globalization.Extensions.builds

(The configs here are just to help VS as the comment says ... they do not affect what we ship.)

Comment (tarekgh) [inline src/System.Globalization.Extensions/src/System.Globalization.Extensions.csproj]: ah, sorry I misunderstood your original comment. I fixed it.

File: src/System.Globalization.Extensions/src/System.Globalization.Extensions.csproj [hunk 6]
```diff
@@ -18,49 +18,17 @@
     <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
```
Comment (weshaggard) [inline src/System.Globalization.Extensions/src/System.Globalization.Extensions.csproj]: Please remove the default Configuration groups above. There is no longer a Windows config.

**Comment by tarekgh** [comment]
@jkotas to let you know I reverted the change in the .builds file because it caused different breaks. But I'll look more on that separately.

**Pull request merged**